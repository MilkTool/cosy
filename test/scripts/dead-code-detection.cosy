fn f() {
  print 'before return'
  if (false) return _
  print 'after first return'
  return _
  print 'after second return'
}

f()

for true {
    print 'before break'
    break
    print 'after break'
}

for 0..3 {
    print 'before continue'
    continue
    print 'after continue'
}
