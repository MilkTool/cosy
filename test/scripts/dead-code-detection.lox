fun f() {
  print "before return";
  if (false) return;
  print "after first return";
  return;
  print "after second return";
}

f();

fun f2() {
    while (true) {
        print "while: before return";
        return;
        print "while: after return";
    }
}

f2();
