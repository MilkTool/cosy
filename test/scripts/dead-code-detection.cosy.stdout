[line 5] Error at "return": Unreachable code after return statement.
[line 13] Error at "break": Unreachable code after return statement.
[line 19] Error at "continue": Unreachable code after return statement.
